test/with_dummyserver/test_proxy_poolmanager.py

```diff
@@ -63,21 +72,75 @@ def test_basic_proxy(self):
             r = http.request("GET", "%s/" % self.https_url)
             assert r.status == 200
 
+    @onlyPy3
     def test_https_proxy(self):
+        with proxy_from_url(self.https_proxy_url, ca_certs=DEFAULT_CA) as https:
+            r = https.request("GET", "%s/" % self.https_url)
+            assert r.status == 200
+
+            r = https.request("GET", "%s/" % self.http_url)
+            assert r.status == 200
+
+    @onlyPy3
+    def test_https_proxy_with_proxy_ssl_context(self):
+        proxy_ssl_context = create_urllib3_context()
+        proxy_ssl_context.load_verify_locations(DEFAULT_CA)
+        with proxy_from_url(
+            self.https_proxy_url,
+            proxy_ssl_context=proxy_ssl_context,
+            ca_certs=DEFAULT_CA,
+        ) as https:
+            r = https.request("GET", "%s/" % self.https_url)
+            assert r.status == 200
+
+            r = https.request("GET", "%s/" % self.http_url)
+            assert r.status == 200
+
+    @onlyPy2
+    def test_https_proxy_not_supported(self):
+        with proxy_from_url(self.https_proxy_url, ca_certs=DEFAULT_CA) as https:
+            r = https.request("GET", "%s/" % self.http_url)
+            assert r.status == 200
+
+            with pytest.raises(ProxySchemeUnsupported) as excinfo:
+                https.request("GET", "%s/" % self.https_url)
+
+            assert "is not supported in Python 2" in str(excinfo.value)
+
+    @withPyOpenSSL
+    @onlyPy3
+    def test_https_proxy_pyopenssl_not_supported(self):
+        with proxy_from_url(self.https_proxy_url, ca_certs=DEFAULT_CA) as https:
+            r = https.request("GET", "%s/" % self.http_url)
+            assert r.status == 200
+
+            with pytest.raises(ProxySchemeUnsupported) as excinfo:
+                https.request("GET", "%s/" % self.https_url)
+
+            assert "isn't available on non-native SSLContext" in str(excinfo.value)
+
+    @onlySecureTransport
+    @onlyPy3
+    def test_https_proxy_securetransport_not_supported(self):
         with proxy_from_url(self.https_proxy_url, ca_certs=DEFAULT_CA) as https:
             r = https.request("GET", "%s/" % self.http_url)
             assert r.status == 200
 
-            with pytest.raises(ProxySchemeUnsupported):
+            with pytest.raises(ProxySchemeUnsupported) as excinfo:
                 https.request("GET", "%s/" % self.https_url)
 
+            assert "isn't available on non-native SSLContext" in str(excinfo.value)
+
+    def test_https_proxy_insecure(self):
```

Comment:
This testcase name might need a refresh?

Reply:
Oops, I'll update and grep around to make sure I'm not missing more.